.github/workflows/release.yml

```diff
@@ -10,6 +10,7 @@ jobs:
   release:
     permissions:
       contents: write # to create release (changesets/action)
+      id-token: write # OpenID Connect token needed for provenance
```

Comment:
what powers exactly does this token grant. is it possible to restrict it only to those required for provenance to work?

see https://docs.github.com/en/actions/security-for-github-actions/security-hardening-your-deployments/about-security-hardening-with-openid-connect#customizing-the-token-claims

cc @pngwn , is this something that needs to be protected from extraction too?